lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,302 @@
+from collections import namedtuple
+import os
+from pathlib import Path
+
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import UnpackTarballs
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
+    min_size, max_size, tar_size = 0, 1000, 586
+
+    def mock_stat(self, *args, follow_symlinks=False):
+        """Mocked replacement for Path.stat()"""
+        StatResult = namedtuple("StatResult", ["st_mode", "st_size"])
+        return StatResult(st_mode=16893, st_size=self.tar_size)
+
+    @pytest.mark.parametrize(
+        "min_size, max_size, total", [(586, 1000, 2), (0, 1000, 2), (0, 586, 0)]
+    )
+    def test_tarball_size(
+        self, monkeypatch, server_config, make_logger, caplog, min_size, max_size, total
+    ):
```

Comment:
`server_config` is required for `server_config.ARCHIVE` value.

Reply:
`server_config` is already a provided mock for tests created in `conftest.py`, this is not a real config object.

Reply:
`server_config` is a fixture which (I'm pretty sure...) creates a temporary, on-disk configuration file which is fed to the real Pbench Server code to instantiate a real config object.

I would like to see the unit tests move away from depending on the file system, among other things.  So, adding new references to the `server_config` fixture as it currently stands would be a move in the wrong direction.

While we could entertain the possibility of updating the `server_config` fixture to properly mock a Server configuration without, e.g., using the file system, that creates a maintenance hassle going forward; moreover, the requirements of the CUT on the config object tend to be so narrow that it would probably be simpler for each test to just mock it to suit the test's specific needs.

Reply:
Are you considering mocking the server configuration object, or should I resolve this conversation?